### ISSUE TITLE
[WAITING ON CHECKLIST - RUSHAT] [$250] Desktop - Categories - App crashes after importing categories and clicking Got it

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#48540]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.29-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**Issue was found when executing this PR:** https://github.com/Expensify/App/pull/47827
**Email or phone of affected tester (no customers):**  applausetester+kh010901@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Launch New Expensify desktop app.
2. Go to workspace settings > Categories.
3. Click 3-dot menu.
4. Click Import spreadsheet.
5. Import the following csv file. (app only crashes with this file)
6. Map Name, Enabled and GL code.
7. Click Import.
8. Click Got it.

## Expected Result:
App will not crash.

## Actual Result:
App crashes after importing categories and clicking Got it.
This issue is only reproducible on desktop app.

## Workaround:
Unknown

## Platforms:

- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

[0409_1.txt](https://github.com/user-attachments/files/16869077/0409_1.txt)
[Bug6592389_1725413369885!--1725413225174.txt](https://github.com/user-attachments/files/16869078/Bug6592389_1725413369885.--1725413225174.txt)
[Bug6592389_1725413369824!Independent.-.Multi.Level.tags (1).csv](https://github.com/user-attachments/files/16869079/Bug6592389_1725413369824.Independent.-.Multi.Level.tags.1.csv)

https://github.com/user-attachments/assets/b0039dde-9175-4c20-8ef1-83640f864705

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021831378368788118300</li>
        <li>Upwork Job ID: 1831378368788118300</li>
        <li>Last Price Increase: 2024-09-04</li>
<li>Automatic offers: </li>
<ul>
<li>nkdengineer | Contributor | 103821693</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @jliexpensify</details>

**Comment by mountiny** [comment]
@filip-solecki @rlinoz @rushatgabhane a proper deploy blocker

**Comment by mountiny** [comment]
Lets try to catch the error with parsing the file and show error message to the user that the file is not supported instead of crashing the app

**Comment by filip-solecki** [comment]
Hi! I am Filip from SWM and I'd like to work on this!

**Comment by mountiny** [comment]
@filip-solecki says this is because the categories are array at first, and the onyx merge causes this error. I have created a web fix to make sure the categories are not empty array but empty object https://github.com/Expensify/Web-Expensify/pull/43372

**Comment by m-natarajan** [comment]
Tester is still able to reproduce 

https://github.com/user-attachments/assets/18f0b618-5fb3-4375-974c-18516494dc2b

**Comment by mountiny** [comment]
@filip-solecki seems like there must be some other root cause for this issue as we are now returning the categories as empty object

Looking for proposals

**Comment by mountiny** [comment]
We might need to put this behind beta if we cant figure this out ahead of the deploy

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
App crashes after importing categories and clicking Got it.
This issue is only reproducible on desktop app.

### What is the root cause of that problem?
When we click on got it button, we will clear the spreadsheet data and it can be done before we navigate to the categories page. Then the app crash here because `spreadsheet?.columns` is `undefined`.

https://github.com/Expensify/App/blob/cb2bdf28addd44c3f0c265f55b8df948c2f9cfc8/src/components/ImportColumn.tsx#L159
### What changes do you think we should make in order to solve the problem?
Safely access here

```
isSelected: spreadsheet?.columns?.[columnIndex] === item.value,
```

https://github.com/Expensify/App/blob/cb2bdf28addd44c3f0c265f55b8df948c2f9cfc8/src/components/ImportColumn.tsx#L159

Or We can wrap the `closeImportPage` in `InteractionManger` or call `Modal.close` callback

https://github.com/Expensify/App/blob/cb2bdf28addd44c3f0c265f55b8df948c2f9cfc8/src/pages/workspace/categories/ImportedCategoriesPage.tsx#L135

### What alternative solutions did you explore? (Optional)

**Comment by rushatgabhane** [comment]
let's assign @nkdengineer ?

**Comment by roryabraham** [comment]
The fact that this can be fixed by adding an optional chain operator, and that wasn't automatically detected, suggests to me that we have incorrect types somewhere

**Comment by mountiny** [comment]
Thanks @nkdengineer what is your eta for the pr?

**Comment by nkdengineer** [comment]
@mountiny The PR is ready.

**Comment by rushatgabhane** [comment]
> The fact that this can be fixed by adding an optional chain operator, and that wasn't automatically detected, suggests to me that we have incorrect types somewhere


We're trying to access a value in `columns: Record<number, string>` using a dynamic key -> `columns[columnIndex]`.

The type is correct. This code is just unsafe in TS. The problem here is that we should ideally have typecheck catch this

**Comment by jliexpensify** [comment]
Payment Summary

- C: @nkdengineer $250
- C+: @rushatgabhane $250 (New.Dot payment) - waiting on checklist

[Upwork job](https://www.upwork.com/jobs/~021831378368788118300)

**Comment by jliexpensify** [comment]
Paid via Upworks and job closed

**Comment by rushatgabhane** [comment]
Hello 👋 

I've been overpaid $250 by mistake in another issue. 

So please don't pay me for this issue. Thank you for understanding

**Comment by jliexpensify** [comment]
Cool, all good - will remove this from the summary! Thanks!

**Comment by jliexpensify** [comment]
@rushatgabhane just waiting on a checklist, if one is needed

**Comment by rushatgabhane** [comment]
not needed here. it is covered by existing regression steps
